ADRs: 017-019 - Adding Impact Sections

tjohnson7021 opened the pull request:
# Description

This PR covers filling in the impact sections for ADRs 017-019 so that it is understood why the decision was made at the time and what the concerns were.

This PR errs on the side of being more verbose than not so that unneeded items can be stripped away during review, so please read carefully.


## Issue #1247 

## Checklist

- [x] I have updated the documentation accordingly

Commit:
WIP: 017 - added empty Impact section

Commit:
017 - added impact section

Commit:
018 - added impact section

Commit:
019 - added impact section

GilmoreA6 commented on adr/018-sql-migrations.md:
```diff
@@ -13,6 +13,47 @@ Accepted
+
+
+- **GitHub Action Reliability:** Relying on prebuilt GitHub Actions introduces a dependency on external tools, which may have compatibility or update issues over time.
+
```
consider adding a risk here regarding changelog structure, as it can affect ability to make changes when one table is referencing another such as our message_link table referencing the received_message_id column in the metadata table (relevant liquibase documentation regarding structure https://docs.liquibase.com/start/design-liquibase-project.html)

GilmoreA6 reviewed (approved):
Added two comments to consider on risk sections of 018 and 019, but otherwise looks good

pluckyswan commented on adr/017-connection-pooling.md:
```diff
@@ -13,6 +13,53 @@ Accepted
+### Risks
+
+- **Connection Leaks:** Improperly handled connections could lead to resource exhaustion, affecting application stability. 
+- Incorrect Configuration: Poorly tuned settings may result in underutilized or overburdened pools. 
```
Missing bold and new line.

pluckyswan reviewed (approved):
Besides my nitpick and Gilmore's comments, LGTM!

jbiskie commented on adr/017-connection-pooling.md:
```diff
@@ -13,6 +13,53 @@ Accepted
+- **Operational Challenges:** Monitoring and maintaining the pool in production environments may require additional effort. 
+
+
+- **Dependency Lock-in:** Switching to another connection pooling library in the future may require significant refactoring.
```
Should our use of OEA reduce this to limited refactoring?

tjohnson7021 replied:
We aren't including mitigations for these ADRs, but if we did, I would definitely be including this answer. I think the fact that we recognize (and document) that we are aware of this risk resolves this story.

jorg3lopez reviewed (approved):
Looks good.

Commit:
Merge branch 'main' into devex/1247-add_impact_sections_to_adrs_017-019

Commit:
017 - fixed formatting

Commit:
018 - added recommended changelog structure issues

Commit:
Revert "018 - added recommended changelog structure issues"

This reverts commit 7728b0f5c18e9a27b02a1edefe494af6273f935a.

Commit:
Updated 018 - updated risks

- Changelog Structure and Dependencies

Commit:
Updated 019 - added suggested risks

- added suggested risk language in regards to adding another table increases potential rollback complexity

Commit:
Merge branch 'main' into devex/1247-add_impact_sections_to_adrs_017-019
